css/src/components/form/checkbox.scss

```diff
@@ -116,4 +142,26 @@ $checkbox-spacing: $spacer-3 !default;
 			border-color: $checkbox-check-color;
 		}
 	}
+
+	/* stylelint-disable no-descending-specificity */
+
+	&.checkbox-danger input[type='checkbox']:not(:checked) {
+		+ .checkbox-check {
+			border-color: $checkbox-danger-border-color;
+		}
+
+		&:focus-visible + .checkbox-check {
+			box-shadow: $checkbox-focus-box-shadow-size $checkbox-danger-border-color;
+		}
+	}
+
+	&.checkbox-success input[type='checkbox']:not(:checked) {
+		+ .checkbox-check {
+			border-color: $checkbox-success-border-color;
+		}
+
+		&:focus-visible + .checkbox-check {
```

Comment:
for focus visible, I believe we have a mixin.

Reply:
updated